scripts/accounts-invariants/pallet-balances-maintenance.py

```diff
@@ -599,6 +617,127 @@ def set_providers_counter_to_one(chain_connection, account_info, internal_system
     return fixed_account_info_hexstring
 
 
+def query_contract_and_code_owners_accounts(chain_connection):
+    """
+    Returns contract accounts and code owners.
+    """
+    code_owners = set()
+    contract_accounts = set()
+
+    log.info(f"Querying code owners.")
+    code_info_of_query = chain_connection.query_map('Contracts', 'CodeInfoOf', page_size=1000)
+
+    for (i, (account_id, info)) in enumerate(code_info_of_query):
+        code_owners.add(info.serialize()['owner'])
+        if i % 5000 == 0 and i > 0:
+            log.info(f"Checked {i} code owners")
+    log.info(f"Total code owners is {len(code_owners)}")
+    log.debug(f"Code owners: {code_owners}")
+
+    log.info(f"Querying contract accounts.")
+    contract_info_of_query = chain_connection.query_map('Contracts', 'ContractInfoOf', page_size=1000)
+    for (i, (account_id, info)) in enumerate(contract_info_of_query):
+        contract_accounts.add(account_id.value)
+        if i % 5000 == 0 and i > 0:
+            log.info(f"Checked {i} contracts")
+    log.info(f"Total contracts count is {len(contract_accounts)}")
+    log.debug(f"Contract accounts: {contract_accounts}")
+
+    return code_owners, contract_accounts
+
+
+def query_accounts_with_consumers_counter_underflow(chain_connection):
+    """
+    Queries all accounts that have an underflow of consumers counter, which is either of those account categories:
+    * `consumers`  == 0, `reserved`  > 0
+    * `consumers`  == 1, `balances.Locks` contain an entry with `id`  == `vesting`
+    * `consumers`  == 2, `balances.Locks` contain an entry with `id`  == `staking`
+    * `consumers`  == 3, `balances.Locks` contain entries with `id`  == `staking` and accountId is in `session.nextKeys`
+    """
+    log.info("Querying session.nextKeys.")
+    next_keys = []
+    for account_id, _ in chain_connection.query_map("Session", "NextKeys", max_results=1000):
```

Comment:
Why do we use `max_results`?

Reply:
Good spot, I copied from docs and did not pick up this, I should be using `page_size`